### PR TITLE
Document clock types

### LINKED ISSUE #552: No documentation for clock types
## Feature request

Please add documentation for the different clock types, i.e. RCL_STEADY_TIME, RCL_SYSTEM_TIME

#### Feature description
There's no documentation about the differences between these times. I'm guessing that STEADY_TIME is monotonically increasing just based on the name, but it would be really nice to have actual documentation. Something that gets into the doxygen docs would be great.

### PR DESCRIPTION
**Pull request by jacobperron**
Resolves #552

### CODE REVIEW

File: rcl/include/rcl/time.h [hunk 1]
```diff
@@ -46,6 +46,14 @@ typedef rcutils_time_point_value_t rcl_time_point_value_t;
 typedef rcutils_duration_value_t rcl_duration_value_t;
 
 /// Time source type, used to indicate the source of a time measurement.
+/**
+ * RCL_ROS_TIME will report the latest value reported by a ROS time source, or
```
Comment (clalancette) [inline rcl/include/rcl/time.h]: I'm not sure if this is the appropriate place, but should we describe what a "ROS time source" is?  I'm not sure everyone will be familiar with it.

Comment (tfoote) [reply rcl/include/rcl/time.h]: Maybe just link to the design document.

Comment (jacobperron) [reply rcl/include/rcl/time.h]: I've added a link to the design doc: c20cafe

File: rcl/include/rcl/time.h [hunk 2]
```diff
@@ -46,6 +46,16 @@ typedef rcutils_time_point_value_t rcl_time_point_value_t;
...
+ * if a ROS time source is not active it reports the same as RCL_SYSTEM_TIME.
+ * For more information about ROS time sources, refer to the design document:
+ * http://design.ros2.org/articles/clock_and_time.html
+ *
+ * RCL_SYSTEM_TIME reports the same value as the system clock.
+ *
+ * RCL_STEADY_TIME reports a value from a monotionically increasing clock.
```
Comment (hidmic) [inline rcl/include/rcl/time.h]: @jacobperron nit: 
```suggestion
 * RCL_STEADY_TIME reports a value from a monotonically increasing clock.
```

Comment (jacobperron) [reply rcl/include/rcl/time.h]: Fixed 97382ba

**Comment by jacobperron** [comment]
Linux: [Build Status](https://ci.ros2.org/job/ci_linux/9494/)

**Pull request merged**